packages/fxa-settings/src/models/Account.ts

```diff
@@ -1046,4 +1050,21 @@ export class Account implements AccountData {
     firefox.accountDeleted(this.uid);
     Storage.factory('localStorage').clear();
   }
+
+  async resetPasswordWithRecoveryKey(
+    accountResetToken:string,
+    password:string,
+    recoveryKeyId:string,
+    kB:string,
+    emailToHashWith:string
```

Comment:
Q: is it expected that the argument orders are different [compared to `this.authClient.resetPasswordWithRecoveryKey()` impl below]?
```jsx
    this.authClient.resetPasswordWithRecoveryKey(
      accountResetToken, // (same)
      emailToHashWith,   // 5th above
      password,          // 2nd above
      recoveryKeyId,     // 3rd above
      { kB }             // 4th above
    )
```

Reply:
This could be confusing / error prone. Let me see what I can do.

Reply:
Alright, it's been updated to a opts object with specified types, this more resilient, and the invocation is pretty clean since it aligns with our model field names.